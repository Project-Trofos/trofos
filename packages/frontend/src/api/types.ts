export type Project = {
  id: number;
  pname: string;
  pkey: string | null;
  description: string | null;
  course_id: number | null;
  public: boolean;
  created_at: string;
  course?: Course;
  is_archive: boolean | null;
};

export type Course = {
  id: number;
  code: string;
  startYear: number;
  startSem: number;
  endYear: number;
  endSem: number;
  cname: string;
  description: string | null;
  public: boolean;
  created_at: string;
  shadow_course: boolean;
  is_archive: boolean | null;
};

export type User = {
  user_email: string;
  user_display_name: string;
  user_id: number;
  projects: Project[];
  courses: CourseRoles[];
  basicRoles: BasicRoles[];
};

export type BasicRoles = {
  user_email: string;
  role_id: number;
};

export type CourseRoles = {
  id: number;
  user_email: string;
  role_id: number;
  course_id: number;
};

export type CreateUserRequest = {
  userEmail: string;
  newPassword: string;
};

export type RegisterUser = {
  userEmail: string;
  newPassword: string;
  userDisplayName: string;
};

export type UserCourseRoleRequest = {
  id: number;
  userRole: number;
  userId: number;
};

export type Role = {
  role_name: string;
  id: number;
};

export type Action = {
  action: string;
};

export type ActionOnRole = {
  id: number;
  action: string;
};

export type ActionsOnRoles = {
  id: number;
  role_name: string;
  actions: Action[];
};

export type ProjectData = Project & {
  course: Course | null;
  users: UserData[];
  sprints: {
    id: number;
    name: string;
  }[];
  backlogStatuses: Omit<BacklogStatusData, 'project_id'>[];
  telegramChannelLink: string;
  owner_id: number | null;
};

export type UserData = {
  user: Pick<User, 'user_id' | 'user_email' | 'user_display_name' | 'courses'>;
};

export type ScrumBoardUserData = {
  user: {
    user_id: number | null;
    user_display_name: string;
    user_email: string;
  };
};

export type Milestone = {
  id: number;
  name: string;
  start_date: string;
  deadline: string;
  course_id: number;
  created_at: string;
};

export type Announcement = {
  id: number;
  title: string;
  content: string;
  user_id: number;
  created_at: string;
  updated_at?: string;
};

// CourseData type returned by BE api. We perform transformResponse to convert the data into type CourseData
export type CourseDataResponse = Omit<CourseData, 'users'> & {
  courseRoles: UserData[];
};

export type CourseData = Course & {
  milestones: Milestone[];
  announcements: Announcement[];
  users: UserData[];
};

export type UserOnRolesOnCourse = {
  id: number;
  user_id: number;
  role_id: number;
  course_id: number;
  role: {
    id: number;
    role_name: string;
  };
};

export type BacklogStatusData = {
  name: string;
  type: 'todo' | 'in_progress' | 'done';
  project_id: number;
  order: number;
};

export type Comment = {
  comment_id: number;
  backlog_id: number;
  project_id: number;
  commenter_id: number;
  content: string;
  created_at: string;
  updated_at: string | null;
  commenter: {
    created_at: string;
    project_id: string;
    user: {
      user_id: number;
      user_email: string;
      user_display_name: string;
    };
  };
};

export type CommentFieldsType = {
  projectId: number;
  backlogId: number;
  commenterId: number;
  content: string;
};

export enum BacklogStatus {
  TODO = 'To do',
  DONE = 'Done',
}

export type BacklogPriority = 'very_high' | 'high' | 'medium' | 'low' | 'very_low' | null;

export type Backlog = {
  backlog_id: number;
  summary: string;
  type: 'story' | 'task' | 'bug';
  priority: BacklogPriority;
  reporter_id: number;
  assignee_id: number | null;
  sprint_id: number | null;
  points: number | null;
  description: string | null;
  project_id: number;
  status: BacklogStatus | string;
  assignee: {
    created_at: string;
    project_id: number;
    user_id: number;
    user: {
      user_email: string;
      user_display_name: string;
    };
  } | null;
  epic_id?: number | null;
  epic?: Epic | null;
};

export type Epic = {
  epic_id: number;
  project_id: number;
  name: string;
  description: string | null;
};

export type DefaultBacklog = {
  backlog_id?: number;
  summary?: string;
  type?: 'story' | 'task' | 'bug';
  priority?: BacklogPriority;
  reporter_id?: number;
  assignee_id?: number | null;
  sprint_id?: number | null;
  points?: number | null;
  description?: string | null;
  project_id?: number;
  status?: BacklogStatus | string;
  assignee?: {
    created_at: string;
    project_id: number;
    user_id: number;
    user: {
      user_email: string;
      user_display_name: string;
    };
  } | null;
  epic_id?: number | null;
};

export type BacklogUpdatePayload = {
  projectId: number;
  backlogId: number;
  srcSprintId?: number | null;
  fieldToUpdate: Partial<Backlog>;
};

export enum BacklogHistoryType {
  CREATE = 'create',
  UPDATE = 'update',
  DELETE = 'delete',
}

export type BacklogHistory = {
  history_type: BacklogHistoryType;
  date: string;
} & Omit<Backlog, 'assignee' | 'summary' | 'description'>;

export type UpdateUserRolePayload = {
  userId: number;
  newRoleId: number;
};

export type ProjectGitLink = {
  project_id: number;
  repo: string;
};

export type ProjectGitLinkData = {
  projectId: number;
  repoLink: string;
};

export type ProjectUserSettings = {
  project_id: number;
  user_id: number;
  email_notification: boolean;
};

export type Settings = {
  current_year: number;
  current_sem: number;
};

export type RetrospectiveVote = {
  id: number;
  retro_id: number;
  user_id: number;
  type: RetrospectiveVoteType;
};

export type Retrospective = {
  id: number;
  sprint_id: number;
  content: string;
  type: RetrospectiveType;
  score: number;
  votes: RetrospectiveVote[];
  is_action_taken: boolean;
};

export enum RetrospectiveType {
  POSITIVE = 'positive',
  NEGATIVE = 'negative',
  ACTION = 'action',
}

export enum RetrospectiveVoteType {
  UP = 'up',
  DOWN = 'down',
}

export type CourseImportCsvPayload = {
  courseId: number;
  payload: FormData;
};

export type Feedback = {
  id: number;
  user_id?: number;
  user: {
    user_email: string;
    user_display_name: string;
  };
  sprint_id: number;
  content: string;
  created_at: string;
  updated_at?: string;
};

export type OAuth2Payload = {
  code: string;
  state: string;
  callbackUrl: string;
};

export type Invite = {
  project_id: number;
  email: string;
  expiry_date: Date;
};

export type UserApiKey = {
  id: number;
  user_id: number;
  api_key: string;
  created_at: string;
  last_used: string | null;
  active: boolean;
};

export type UserGuideQueryResponse = {
  answer: string;
  links: Array<string>;
};

export type FeatureFlag = {
  feature_name: string;
  active: boolean;
};

export type ProjectAssignment = {
  id: number;
  targetProjectId: number;
  sourceProjectId: number;
};
