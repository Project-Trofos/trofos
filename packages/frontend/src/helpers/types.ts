export type PickRename<T, K extends keyof T, R extends PropertyKey> = Pick<Omit<T, K> & { [P in R]: T[K] }, R>;
