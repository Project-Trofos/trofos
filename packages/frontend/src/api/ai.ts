import trofosApiSlice from '.';
import { UserGuideQueryResponse, UserGuideRecommendation } from './types';

const extendedApi = trofosApiSlice.injectEndpoints({
  endpoints: (builder) => ({
    answerUserGuideQuery: builder.mutation<UserGuideQueryResponse, { query: string; isEnableMemory: boolean }>({
      query: ({ query, isEnableMemory }) => ({
        url: `ai/userGuideQuery`,
        credentials: 'include',
        method: 'POST',
        body: { query, isEnableMemory },
      }),
    }),
    recommendUserGuideSections: builder.mutation<UserGuideRecommendation[], void>({
      query: () => ({
        url: `ai/recommendUserGuide`,
        method: 'POST',
        credentials: 'include',
      }),
    }),
  }),
});

export const { useAnswerUserGuideQueryMutation, useRecommendUserGuideSectionsMutation } = extendedApi;
