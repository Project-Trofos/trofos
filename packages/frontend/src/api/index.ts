import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

// Define a service using a base URL and expected endpoints
const trofosApiSlice = createApi({
  reducerPath: 'trofosApi',
  baseQuery: fetchBaseQuery({
    baseUrl: process.env.NODE_ENV === 'test' ? 'http://localhost:3001/api' : '/api',
    prepareHeaders: (headers) => {
      headers.set('Cache-control', 'no-cache');
      return headers;
    },
  }),
  tagTypes: [
    'Project',
    'UserInfo',
    'Course',
    'Backlog',
    'BacklogHistory',
    'User',
    'Sprint',
    'ActionsOnRoles',
    'CourseRoles',
    'ProjectRoles',
    'Settings',
    'StandUpHeader',
    'StandUp',
    'Retrospective',
    'ProjectUserSettings',
    'Feedback',
  ],
  endpoints: () => ({}),
});

// Define a service using a base URL and expected endpoints
export const nusmodsApiSlice = createApi({
  reducerPath: 'nusmodsApi',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://api.nusmods.com/v2/2022-2023' }),
  endpoints: () => ({}),
});

export default trofosApiSlice;
