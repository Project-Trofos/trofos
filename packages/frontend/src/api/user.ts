/* eslint-disable import/prefer-default-export */
import trofosApiSlice from '.';
import { User, CreateUserRequest } from './types';

const extendedApi = trofosApiSlice.injectEndpoints({
  endpoints: (builder) => ({
    getUsers: builder.query<User[], void>({
      query: () => ({
        url: 'user/',
        credentials: 'include',
      }),
      providesTags: ['User'],
    }),
    findUserByEmail: builder.mutation<Pick<User, 'user_email'>, string>({
      query: (userEmail) => ({
        url: `user/${userEmail}`,
        method: 'GET',
        credentials: 'include',
      }),
    }),
    createUser: builder.mutation<void, CreateUserRequest>({
      query: (user) => ({
        url: 'user/',
        method: 'POST',
        body: {
          userEmail: user.userEmail,
          newPassword: user.newPassword,
        },
        credentials: 'include',
      }),
      invalidatesTags: ['User'],
    }),
  }),
  overrideExisting: false,
});

export const { useGetUsersQuery, useFindUserByEmailMutation, useCreateUserMutation } = extendedApi;
