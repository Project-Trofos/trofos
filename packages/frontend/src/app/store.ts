import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';
import trofosApiSlice, { nusmodsApiSlice } from '../api';

const store = configureStore({
  reducer: {
    [trofosApiSlice.reducerPath]: trofosApiSlice.reducer,
    [nusmodsApiSlice.reducerPath]: nusmodsApiSlice.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(trofosApiSlice.middleware).concat(nusmodsApiSlice.middleware),
});

// optional, but required for refetchOnFocus/refetchOnReconnect behaviors
// see `setupListeners` docs - takes an optional callback as the 2nd arg for customization
setupListeners(store.dispatch);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
