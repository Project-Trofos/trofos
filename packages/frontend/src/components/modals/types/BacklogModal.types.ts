export interface BacklogFormFields extends FormData {
  projectId: number;
}

export type BacklogSelect = {
  id: string;
  name: string;
};
