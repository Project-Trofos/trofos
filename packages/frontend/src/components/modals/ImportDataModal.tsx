import React, { useState } from "react";
import { UploadOutlined } from '@ant-design/icons';
import { Button, Modal, message, Upload, Space } from 'antd';
import type { RcFile, UploadFile, UploadProps } from 'antd/es/upload/interface';
import { useImportCsvMutation } from "../../api/course";
import { getErrorMessage } from "../../helpers/error";
import { CourseData, ProjectData } from '../../api/types';

const CSV_TEMPLATE_PATH = "/download/importCourseData.csv";
const CSV_TEMPLATE_NAME = "importCourseData.csv";

export default function ImportDataModal({ course, projects } : { course: CourseData | undefined, projects: ProjectData[] | undefined }) : JSX.Element {

    const [isModalOpen, setIsModalOpen] = useState(false);
    const [fileList, setFileList] = useState<UploadFile[]>([]);
    const [isUploading, setIsUploading] = useState(false);
    const [importCsv] = useImportCsvMutation();

    const showModal = () => {
        setIsModalOpen(true);
    };

    const handleOk = async () => {
        if (!projects || projects.length > 0) {
            message.error("Course already has projects. Please delete them before uploading a new CSV file.");
        } else {
            const formData = new FormData();
            formData.append('file', fileList[0] as RcFile);
            setIsUploading(true);
            try {
                if (course) {
                    await importCsv({
                        courseId: course.id,
                        payload: formData
                    }).unwrap();
                    setFileList([]);
                    message.success('Import successful.');
                }
            } catch (error) {
                message.error(getErrorMessage(error));
            }
            setIsUploading(false);
        }
        setIsModalOpen(false);
    };
    
    const handleCancel = () => {
        setIsModalOpen(false);
    };

    const props: UploadProps = {
        onRemove: (file) => {
          const index = fileList.indexOf(file);
          const newFileList = fileList.slice();
          newFileList.splice(index, 1);
          setFileList(newFileList);
        },
        beforeUpload: (file) => {
          setFileList([file]);
    
          return false;
        },
        fileList,
    };

    return (
    <>
        <Button type="primary" onClick={showModal}>
          Import CSV Data
        </Button>
        <Modal 
            title="Import CSV Data" 
            okButtonProps={{
                disabled: fileList.length === 0,
                loading: isUploading
            }} 
            okText={isUploading ? 'Uploading' : 'Import'}
            open={isModalOpen} 
            onOk={handleOk} 
            onCancel={handleCancel}
        >
            <Space direction="vertical">
                {/* eslint-disable-next-line react/jsx-props-no-spreading */}
                <Upload {...props}>
                        <Button icon={<UploadOutlined />}>Select File</Button>
                </Upload>
                <a href={CSV_TEMPLATE_PATH} download={CSV_TEMPLATE_NAME}>Download csv template</a>
            </Space>
        </Modal>
      </>
    )
}