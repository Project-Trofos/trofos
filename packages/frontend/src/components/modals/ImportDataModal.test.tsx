import React from 'react';
import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import user from '@testing-library/user-event';
import server from '../../mocks/server';
import ImportDataModal from './ImportDataModal';
import store from '../../app/store';
import { CourseData, ProjectData } from '../../api/types';

const mockFile = new File(['hello'], 'hello.png', { type: 'image/png' });

const mockCourseData: CourseData = {
  id: 1,
  code: 'course_id',
  startYear: 2022,
  startSem: 10,
  endYear: 2022,
  endSem: 10,
  cname: 'course1',
  description: 'project1_description',
  public: false,
  created_at: '2022-09-15T01:58:01.735Z',
  milestones: [],
  announcements: [],
  shadow_course: false,
  users: [
    {
      user: {
        user_display_name: 'user',
        user_id: 1,
        user_email: 'email',
        courses: [
          {
            id: 1,
            user_email: 'email',
            role_id: 1,
            course_id: 1,
          },
        ],
      },
    },
  ],
};

const mockProjectData: ProjectData = {
  id: 1,
  pname: 'c1',
  created_at: new Date().toISOString(),
  course_id: 5,
  course: mockCourseData,
  pkey: null,
  description: 'd1',
  public: false,
  users: [],
  sprints: [],
  backlogStatuses: [],
};

describe('test import data modal', () => {
  // Establish API mocking before all tests.
  beforeAll(() => server.listen());

  // Reset any request handlers that we may add during the tests,
  // so they don't affect other tests.

  afterEach(() => server.resetHandlers());

  // Clean up after the tests are finished.
  afterAll(() => server.close());

  const setup = (course: CourseData | undefined, projects: ProjectData[] | undefined) => {
    const { baseElement, debug } = render(
      <Provider store={store}>
        <ImportDataModal course={course} projects={projects} />
      </Provider>,
    );

    // Open modal
    const button = screen.getByText(/import csv data/i);
    fireEvent.click(button);

    return { baseElement, debug };
  };

  it('should display the components correctly', () => {
    const { baseElement } = setup(mockCourseData, [mockProjectData]);

    // Ensure components are present
    expect(screen.getByText(/select file/i)).toBeInTheDocument();
    expect(screen.getByText(/download csv template/i)).toBeInTheDocument();
    expect(screen.getByText(/cancel/i)).toBeInTheDocument();
    expect(screen.getByText(/^import$/i)).toBeInTheDocument();

    // Compare with snapshot to ensure structure remains the same
    expect(baseElement).toMatchSnapshot();
  });

  it('should disable the import button when no files are selected', () => {
    setup(mockCourseData, [mockProjectData]);

    const button = screen.getByRole('button', { name: /^import$/i });
    expect(button).toBeDisabled();
  });

  it('should enable the import button when a file is selected to be uploaded', async () => {
    setup(mockCourseData, [mockProjectData]);
    const input = screen.getByTestId('upload-button') as HTMLInputElement;
    /* eslint-disable testing-library/no-unnecessary-act */
    await act(async () => {
      fireEvent.change(input, {target: {files: [mockFile]}});
    })
    /* eslint-enable */

    expect(input.files![0]).toStrictEqual(mockFile);
    expect(input.files).toHaveLength(1);

    const button = screen.getByRole('button', { name: /^import$/i });
    expect(button).toBeEnabled();
  });

  it('should prevent the user from uploading a csv if the course has projects', async () => {

    setup(mockCourseData, [mockProjectData]);
    const input = screen.getByTestId('upload-button') as HTMLInputElement;
    /* eslint-disable testing-library/no-unnecessary-act */
    await act(async () => {
      fireEvent.change(input, {target: {files: [mockFile]}});
    })
    /* eslint-enable */

    expect(input.files![0]).toStrictEqual(mockFile);
    expect(input.files).toHaveLength(1);

    const button = screen.getByText(/^import$/i);
    fireEvent.click(button);
    expect(
      await screen.findByText(/course already has projects. please delete them before uploading a new csv file./i),
    ).toBeInTheDocument();
  });
});
