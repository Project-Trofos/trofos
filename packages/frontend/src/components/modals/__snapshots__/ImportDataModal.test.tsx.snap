// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test import data modal > should display the components correctly 1`] = `
<body>
  <div>
    <button
      class="ant-btn ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
      type="button"
    >
      <span>
        Import CSV Data
      </span>
    </button>
  </div>
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 520px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  Import CSV Data
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <div
                  class="ant-space ant-space-vertical ant-space-gap-row-small ant-space-gap-col-small"
                >
                  <div
                    class="ant-space-item"
                  >
                    <span
                      class="ant-upload-wrapper"
                    >
                      <div
                        class="ant-upload ant-upload-select"
                      >
                        <span
                          class="ant-upload"
                        >
                          <input
                            accept=""
                            data-testid="upload-button"
                            name="file"
                            style="display: none;"
                            type="file"
                          />
                          <button
                            class="ant-btn ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
                            type="button"
                          >
                            <span
                              class="ant-btn-icon"
                            >
                              <span
                                aria-label="upload"
                                class="anticon anticon-upload"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="upload"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                  />
                                </svg>
                              </span>
                            </span>
                            <span>
                              Select File
                            </span>
                          </button>
                        </span>
                      </div>
                      <div
                        class="ant-upload-list ant-upload-list-text"
                      />
                    </span>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <a
                      download="importCourseData.csv"
                      href="/download/importCourseData.csv"
                    >
                      Download csv template
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="ant-modal-footer"
              >
                <button
                  class="ant-btn ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
                  type="button"
                >
                  <span>
                    Cancel
                  </span>
                </button>
                <button
                  class="ant-btn ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
                  disabled=""
                  type="button"
                >
                  <span>
                    Import
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
